consensus/src/recovery_manager.rs

```diff
@@ -0,0 +1,151 @@
+        self.sync_up(sync_info, author).await
+    }
+
+    pub async fn sync_up(&mut self, sync_info: &SyncInfo, peer: Author) -> Result<RecoveryData> {
+        sync_info.verify(&self.epoch_state.verifier)?;
+        ensure!(
+            sync_info.highest_round() > self.last_committed_round,
+            "[RecoveryManager] Received sync info has lower round number than committed block"
+        );
+        ensure!(
+            sync_info.epoch() == self.epoch_state.epoch,
+            "[RecoveryManager] Received sync info is in different epoch than committed block"
+        );
+        let mut retriever = BlockRetriever::new(
+            self.network.clone(),
+            peer,
+            self.epoch_state
+                .verifier
+                .get_ordered_account_addresses_iter()
+                .collect(),
+        );
+        let recovery_data = BlockStore::fast_forward_sync(
+            sync_info.highest_ordered_cert(),
+            sync_info.highest_commit_cert(),
+            &mut retriever,
+            self.storage.clone(),
+            self.state_computer.clone(),
+        )
+        .await?;
+
+        Ok(recovery_data)
+    }
+
+    pub async fn start(
+        mut self,
+        mut event_rx: aptos_channel::Receiver<
+            (Author, Discriminant<VerifiedEvent>),
+            (Author, VerifiedEvent),
+        >,
+        close_rx: oneshot::Receiver<oneshot::Sender<()>>,
+    ) {
+        info!(epoch = self.epoch_state.epoch, "RecoveryManager started");
+        let mut close_rx = close_rx.into_stream();
+        loop {
+            futures::select! {
+                (peer_id, event) = event_rx.select_next_some() => {
+                    let result = match event {
+                        VerifiedEvent::ProposalMsg(proposal_msg) => {
+                            monitor!(
+                                "process_recovery",
+                                self.process_proposal_msg(*proposal_msg).await
+                            )
+                        }
+                        VerifiedEvent::VoteMsg(vote_msg) => {
+                            monitor!("process_recovery", self.process_vote_msg(*vote_msg).await)
+                        }
+                        VerifiedEvent::UnverifiedSyncInfo(sync_info) => {
+                            monitor!(
+                                "process_recovery",
+                                self.sync_up(&sync_info, peer_id).await
+                            )
+                        }
+                        unexpected_event => unreachable!("Unexpected event: {:?}", unexpected_event),
+                    }
+                    .with_context(|| format!("from peer {}", peer_id));
+
+                    match result {
+                        Ok(_) => {
+                            info!("Recovery finishes for epoch {}, RecoveryManager stopped. Please restart the node", self.epoch_state.epoch);
+                            process::exit(0);
```

Comment:
Do we have to restart the node? By the time we restart we might need to sync again. 
Cannot we go back to the epoch manger and create the round manager without restarting?